docs/handling-uuids-on-post-put.md

```diff
@@ -0,0 +1,53 @@
+# Handling UUIDs in `POST` and `PUT` operations
+
+## `POST`
+
+When creating new objects, `POST` is the preferred option. The UUID of the object does
+not need to be specified, so `POST` should be used when there is not a preferred UUID
+for the new object. If the UUID is specified in the body, an HTTP 4xx error will be returned.
+
+## `PUT`
+
+When creating new objects, `PUT` should only be used when a specific UUID is required.
+Additionally, it can be used to update an object by replacing its entire definition with
+ an updated version. Currently, `PUT` is defined to include the UUID in the request path
+ and the request body, however it is not necessary to specify the UUID in the body. If the
+ UUID is specified in both the path and the body, the UUID must match, or else an HTTP 4xx
+ error will be returned. If an object with the specified UUID exists, the `PUT` request will
+ update the existing object. If no objects exist with the specified UUID, a new object will
+ be created with that UUID.
+
```

Comment:
Suggestion: write these as bullets for better clarity. Would be also be easier to read

Reply:
I think we could solve this by keeping the narrative and just moving the relevant diagram to this location

Reply:
@schen2166 I chose not to write it as bullets because it appears that almost all of the documentation written in the repository is written in paragraphs so I wanted to keep it consistent with that. @kylelaker I agree. The diagrams would help reinforce what each section is saying.